x-pack/plugins/observability_solution/apm_data_access/server/utils/unflatten_known_fields.test.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// unflatten_known_fields.test.ts
+import { unflattenKnownApmEventFields } from './unflatten_known_fields';
+
+describe('unflattenKnownApmEventFields', () => {
```

Comment:
ideally this also has tests for:
- not single-value-ing unknown fields
- having known fields override unknown fields on collisions (like `service.name` and `service.name.text`)

Reply:
done! thanks.

Reply:
> not single-value-ing unknown fields

unknown fields are returned by the function, btw. Isn't it how it's supposed to work?

Reply:
yes, but it shouldn't convert an array to a single value

Reply:
Let me know if there are more test cases we need to cover.